### ISSUE TITLE
Disabled markers appear after multiple x-axis extremes change with `a11y` and `styledMode` enabled

### ISSUE THREAD

**Issue by jedrzejruta** [highcharts/highcharts#21915]
#### Expected behaviour
The markers don't appear after changing x-axis extremes.

#### Actual behaviour
Markers appear on the line chart even when they are explicitly disabled.

#### Live demo with steps to reproduce
Minimal demo with Highcharts Core and a11y: https://jsfiddle.net/BlackLabel/uvkbnt8c/.

Notice how the markers appear only after the 2nd zoom in/zoom out.

Stock demo with Navigator:
[https://jsfiddle.net/BlackLabel/4r8fLd62/](https://jsfiddle.net/BlackLabel/4r8fLd62/)

Grab left handle of navigator and set it in center, lift the mouse button up and then quickly go back with the handle to the left side. The markers will appear, sometimes leaving a trace of markers. The issue appears only when `styledMode` is set to true and `a11y` module is included.

![stock_accessibility_markers](https://github.com/user-attachments/assets/3ab78188-b527-47fb-8b57-90f6cc552629)

#### Product version
Highcharts v11.4.8

#### Workaround: 
Hide the markers with CSS: [demo](https://jsfiddle.net/BlackLabel/mxa0t8uy/)
`.highcharts-markers {
  display: none;
}
`

**Comment by Blackhol3** [comment]
The bug isn't tied to Highstock specifically; it's happening with the `a11y` module included and zooming enabled. I can't reproduce a minimal example though, so there is a probably something more to it.

As for the workaround, hiding the markers works well, but they are still calculated and added to the DOM. Here is a more drastic solution:
```typescript
Highcharts.wrap(Highcharts.Series.prototype, 'drawPoints', function (this: Highcharts.Series, proceed: (...args: unknown[]) => unknown, ...args: unknown[]) {
	Highcharts.merge(true, this.options, {marker: {enabled: false}});
	return proceed.apply(this, args);
});
```

**Comment by jedrzejruta** [comment]
Hi @Blackhol3!

Thanks for providing more details to this issue and a workaround.

**Reproducible demo:** https://jsfiddle.net/BlackLabel/uvkbnt8c/

_Internal note:_ The issue seems to happen only with data points > 300 & after a 2nd zoom in & zoom out, with the zoom in bigger than half of the x-axis extremes.

**Comment by jedrzejruta** [comment]
Fixed via #23388.
